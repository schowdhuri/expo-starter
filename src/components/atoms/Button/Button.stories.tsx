import type { ComponentStory, ComponentMeta } from "@storybook/react-native";
import { MyButton } from "./Button";

const MyButtonMeta: ComponentMeta<typeof MyButton> = {
  title: "MyButton",
  component: MyButton,
  argTypes: {
    onPress: { action: "pressed the button" },
  },
  args: {
    text: "Hello world",
  },
};

export default MyButtonMeta;

export const Basic: ComponentStory<typeof MyButton> = (args) => (
  <MyButton {...args} />
);
