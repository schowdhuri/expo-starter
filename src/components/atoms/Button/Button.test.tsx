import { fireEvent, render } from "@testing-library/react-native";
import { MyButton } from "./Button";

describe("atoms::Button", () => {
  test("renders without exploding", () => {
    const clickSpy = jest.fn();
    const { toJSON } = render(
      <MyButton onPress={clickSpy} text="Sample button" />
    );
    expect(toJSON()).toMatchSnapshot();
  });

  test("calls onPress", () => {
    const clickSpy = jest.fn();
    const { getByRole } = render(
      <MyButton onPress={clickSpy} text="Sample button" />
    );
    fireEvent.press(getByRole("button"));
    expect(clickSpy).toHaveBeenCalledTimes(1);
  });
});
