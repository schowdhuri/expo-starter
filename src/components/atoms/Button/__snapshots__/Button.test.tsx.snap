// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`atoms::Button renders without exploding 1`] = `
<View
  accessibilityRole="button"
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "violet",
      "opacity": 1,
      "paddingHorizontal": 16,
      "paddingVertical": 8,
    }
  }
>
  <Text
    style={
      Object {
        "color": "black",
      }
    }
  >
    Sample button
     :)
  </Text>
</View>
`;
